### PR TITLE
citations file

### PR DESCRIPTION
**Pull request by keewis**
- [x] follow-up to #80

This adds a `CITATION.cff` file, together with a `pre-commit` hook that validates it. Note that this is an initial draft, so feel free to suggest changes.

### CODE REVIEW

**Comment by tinaok** [comment]
@keewis here are 2 orcid missing;

@benbovy : https://orcid.org/0009-0001-4011-3574
@danlooo :**0000-0002-4024-4443**

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    given-names: "Alejandro"
+    orcid: "https://orcid.org/0000-0002-9264-1539"
+  - family-names: "Strobl"
+    given-names: "Peter"
+    orcid: "https://orcid.org/0000-0003-2733-1822"
+  - family-names: "Fouilloux"
+    given-names: "Anne"
+    orcid: "https://orcid.org/0000-0002-1784-2920"
+  - family-names: "Loos"
+    given-names: "Daniel"
```
Comment (keewis) [inline CITATION.cff:27]:
```suggestion
  - family-names: "Loos"
    given-names: "Daniel"
    orcid: "https://orcid.org/0000-0002-4024-4443"
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,42 @@
+cff-version: 1.2.0
+title: xdggs
+message: "If you use this software, please cite it as below."
+type: software
+authors:
+  - family-names: "Magin"
+    given-names: "Justus"
+    orcid: "https://orcid.org/0000-0002-4254-8002"
+  - family-names: "Bovy"
+    given-names: "Benoît"
```
Comment (keewis) [inline CITATION.cff:10]:
```suggestion
  - family-names: "Bovy"
    given-names: "Benoît"
    orcid: "https://orcid.org/0009-0001-4011-3574"
```

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    given-names: "Peter"
+    orcid: "https://orcid.org/0000-0003-2733-1822"
+  - family-names: "Fouilloux"
+    given-names: "Anne"
+    orcid: "https://orcid.org/0000-0002-1784-2920"
+  - family-names: "Loos"
+    given-names: "Daniel"
+    orcid: "https://orcid.org/0000-0002-4024-4443"
+  - family-names: "Chan"
+    given-names: "Wai Tik"
```
Comment (keewis) [inline CITATION.cff:31]: cc @allixender @tik65536

Comment (tik65536) [reply CITATION.cff:31]: @keewis , Thx. That's correct.

Comment (keewis) [reply CITATION.cff:31]: Do you want to add your orcid (if you have one)? Otherwise I'll merge as-is.

Comment (tik65536) [reply CITATION.cff:31]: O , sorry , here is my OCID: 0009-0005-3779-139X
Thanks,
Tik

Comment (keewis) [reply CITATION.cff:31]:
```suggestion
  - family-names: "Chan"
    given-names: "Wai Tik"
    orcid: "0009-0005-3779-139X"
```

**Pull request merged**